modules/peridynamics/src/mesh/PeridynamicsMesh.C

```diff
@@ -0,0 +1,484 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PeridynamicsMesh.h"
+
+#include "libmesh/elem.h"
+
+registerMooseObject("PeridynamicsApp", PeridynamicsMesh);
+
+template <>
+InputParameters
+validParams<PeridynamicsMesh>()
+{
+  InputParameters params = validParams<MooseMesh>();
+  params.addClassDescription("Mesh class to store and return peridynamics specific mesh data");
+
+  params.addParam<Real>("horizon_radius", "Value of horizon size in terms of radius");
+  params.addParam<Real>("horizon_number",
+                        "The material points spacing number, i.e. ratio of horizon radius to the "
+                        "effective mesh spacing");
+  params.addParam<Real>("ratio_of_horizons",
```

Comment:
This would be better named something like `bond_associated_horizon_ratio` to make it clear what horizon it's referring to.